File: week18/p_42889.py
N = 5
stages = [2, 1, 2, 6, 2, 4, 3, 3]	

def solution(N, stages):
    answer = []

  



print(solution(N,stages))
        



# def solution(N, stages):
#     answer = []
#     failure = [0]*(N+1)
#     for i in range(1,N+1):
#         for j in range(len(stages)):
#             if stages[j] >= i:
#                 failure[i] += 1
#     for i in range(1,N+1):
#         try: 
#             answer.append([stages.count(i)/failure[i],i])

#         except:
#             answer.append([0,i])
       
#     fin = sorted(answer, key = lambda x: -x[0] )
#     real_fin = [x[1] for x in fin]
#     return(real_fin)

File: week19/p_42889.js
stages = [2, 1, 2, 6, 2, 4, 3, 3]	
N = 5

function solution(N,stages){
    let divide = []
    let count = 0

    for (let i = 1; i<=N; i++){
        for (let j = 0; j<stages.length; j++){
            if (stages[j]>= i){
                count += 1
            }

        }
        divide.push([stages.filter(x=>x==i).length / count,i])
        count = 0

    }
divide.sort((a,b)=>{return b[0]-a[0]})

let answer = []
for(let i = 0; i<divide.length; i++){
    answer.push(divide[i][1])
}



return answer
}

console.log(solution(N,stages))